### ISSUE TITLE
Adjusting slider reads out multiple alerts

### ISSUE THREAD

**Issue by liam-mulhall** [phetsims/john-travoltage#408]
### Test Device

MacBook Air

### Operating System

macOS 11.2.3

### Browser

Safari 14.0.3

### Problem Description

For https://github.com/phetsims/QA/issues/645. We noticed that when adjusting the voice pitch slider, it will read out multiple (every?) changes in pitch. So if I just slide it from one end to the other, it reads out a ton of alerts. Same thing happens with the voice speed.

### Steps to Reproduce

Make sure voicing is on and then slide the slider from one end to the other.

<details>
<summary><b>Troubleshooting Information</b></summary>

Name: ‪John Travoltage‬
URL: https://phet-dev.colorado.edu/html/john-travoltage/1.6.0-preferences.4/phet/john-travoltage_en_phet.html
Version: 1.6.0-preferences.4 2021-04-28 22:49:44 UTC
Features missing: touch
User Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0.3 Safari/605.1.15
Language: en-us
Window: 1264x795
Pixel Ratio: 1/1
WebGL: WebGL 1.0
GLSL: WebGL GLSL ES 1.0 (1.0)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 15 uniform: 1024
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 8192x8192
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by jessegreenberg** [comment]
Confirmed, I am also seeing this for the rate slider.

**Comment by jessegreenberg** [comment]
I am actually seeing a collection of alerts happen on safari in a number of situations. If I spam a checkbox, it will repeat the queued alerts rather than speak the latest.

**Comment by jessegreenberg** [comment]
There is a workaround in Safari that delays the webSpeaker from speaking which was found to make the `end` event of the Utterance happen more consistently on that platform. I wonder if that is contributing and can be removed now.

**Comment by jessegreenberg** [comment]
```js
      // on safari, giving a bit of a delay to the speak request makes the `end`
      // SpeechSynthesisUtterance event come through much more consistently
      if ( platform.safari ) {
        stepTimer.setTimeout( () => {
          this.getSynth().speak( speechSynthUtterance );
        }, 500 );
      }
      else {
        this.getSynth().speak( speechSynthUtterance );
      }
```

If I remove that workaround, the problem does seem to go away. But I confirmed that there are situations where we don't get the "ended" event and so certain highlights do not disappear.

**Comment by jessegreenberg** [comment]
I confirmed that the workaround above is improving the consistenty of the `end` event of the SpeechSynthesisUtterance. But I would love to remove it, it makes the output inconsistent and causes this issue. I am going to remove this workaround, this issue and platform inconsistencies are worse than sometimes not getting the end event in my opinion. 

It seems like the failure to send events happens most often when another utterance is interrupted. I tried a different workaround where I wrapped the cancel() with a puase() and resume() in case that helped. It didn't work.

**Comment by jessegreenberg** [comment]
It looks like there is an `error` event that consistently fires when the synth is canceled. I don't see any details about the error, but maybe Safari triggers this instead of the `end` event on a cancel. Perhaps this could be listened in addition to the end event on that platform.

**Comment by jessegreenberg** [comment]
Done in the above commit, this is working pretty well. In Safari, there are times where the highlight will not show up. But since we don't consistently get the event I think there is nothing more that we can do. It is handled more gracefully on our end and will not cause output difference on Safari anymore.
